site/en/docs/extensions/mv3/getstarted/tut-quick-reference/index.md

```diff
@@ -0,0 +1,469 @@
+
+{% endAside %}
+
+### _Optional: Debugging the service worker_ {: #step-3 }
+
+Let's quickly go over how to find the service worker logs and know when it has terminated. First, follow the instructions to [Load an unpacked extension][doc-dev-basics-unpacked]. After 30 seconds it will show "service worker(inactive)" which means the service worker has terminated. 
+
+Click on the "service worker(inactive)" hyperlink to inspect it. 
+
+{% Video src="video/BhuKGJaIeLNPW9ehns59NfwqKxF2/D1XRaA6q4xn9Ylwe1u1N.mp4", width="800", height="314", autoplay="true", muted="true", loop="true" %}
+
+Did you notice that inspecting the service worker woke it up? That's right! Opening the service worker in the devtools will keep it active. This means, if you want to make sure that your extension behaves correctly when your service worker is terminated, remember to close the DevTools.
+
+Now let's break the extension to learn where to locate errors. One way to do this is to delete the ".js" from the `'./sw-omnibox.js'` import in the `service-worker.js` file. Chrome will be unable to register the service worker.
+
+Go back to chrome://extensions and refresh the extension. The following error will appear:
+
+{% Video src="video/BhuKGJaIeLNPW9ehns59NfwqKxF2/AbMNDSbURLKjH1Jm1C9Q.mp4", width="400", height="477", autoplay="true", muted="true", loop="true" %}
+
+See [Debugging extensions](tbd) for more ways debug the extension service worker.
+
+{% Aside 'caution' %}
+Don't forget to fix the file name before moving on!
+{% endAside %}
+
+### Step 4: Initialize the state {: #step-4 }
+
+Extensions can save initial values to storage on installation. To use the [`chrome.storage`][api-storage] API, we need to request permission in the manifest:
+
+{% Label %}manifest.json:{% endLabel %}
+
+```json
+{
+  ...
+  "permissions": ["storage"],
+}
+```
+
+The following code save the default suggestions to storage when the extension is first installed:
+
+{% Label %}sw-omnibox.js:{% endLabel %}
+
+```js
+...
+// Save default API suggestions
+chrome.runtime.onInstalled.addListener(({ reason }) => {
+  if (reason === 'install') {
+    chrome.storage.local.set({
+      apiSuggestions: ['tabs', 'storage', 'scripting']
+    });
+  }
+});
+```
+
+Service workers do not have direct access to the [window object][mdn-window], therefore cannot use
+[window.localStorage()][mdn-local-storage] to store values. Also, service workers are short-lived execution environments;
+they get terminated repeatedly throughout a user's browser session, which makes it incompatible with
+global variables.
+
+See [Saving state](TBD) to learn about storage options for extension service workers.
+
+### Step 5: Register your events {: #step-5 }
+
+All event listeners need to be registered in the global scope of the service worker. In other words, event listeners should not be nested in functions. This way Chrome can immediately invoke all event handlers, even if the extension's async startup logic hasn't finished. 
```

Comment:
Not sure if this is a better way to phrase this:

```suggestion
All event listeners need to be statically registered in the global scope of the service worker. In other words, event listeners should not be nested in async functions. This way Chrome can ensure that all event handlers are restored in case of a service worker reboot.
```

Maybe rephrase to:

"